### PR TITLE
Add override for form states

### PR DESCRIPTION
**Pull request by jadon1979**
## Summary
- *States in Mexico are not properly parsing in ES.  This adds the ability to override the underlying form fields for these cases*

## Related issue(s)
(https://app.zenhub.com/workspaces/10-10-health-apps-5fff0cfd1462b6000e320fc7/issues/gh/department-of-veterans-affairs/va.gov-team/65373)[https://app.zenhub.com/workspaces/10-10-health-apps-5fff0cfd1462b6000e320fc7/issues/gh/department-of-veterans-affairs/va.gov-team/65373]

## Testing done
- *Rspec and manual*

## What areas of the site does it impact?
*1010ez form submission*

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -453,6 +453,7 @@ app/services/facilities/website_url_service.rb @department-of-veterans-affairs/v
 app/services/form1010cg @department-of-veterans-affairs/vfs-10-10 @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
 app/services/form1010cg/auditor.rb @department-of-veterans-affairs/vfs-10-10 @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
 app/services/form1010cg/service.rb @department-of-veterans-affairs/vfs-10-10 @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
+app/services/hca/overrides_parser.rb @department-of-veterans-affairs/vfs-authenticated-experience-backend @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
```
Comment (lihanli) [inline .github/CODEOWNERS:456]: can you change @department-of-veterans-affairs/vfs-authenticated-experience-backend to @department-of-veterans-affairs/vfs-10-10?

**Review by JoshingYou1 — APPROVED** [review]
LGTM! Nice work Chapley!

**Comment by lihanli** [comment]
override should happen in the model after the schema validation is done. if you override it in the controller then the schema validation won't pass

File: .github/CODEOWNERS (hunk 1 above)
Comment (lihanli) [reply .github/CODEOWNERS:456]: can you change https://github.com/orgs/department-of-veterans-affairs/teams/vfs-authenticated-experience-backend to https://github.com/orgs/department-of-veterans-affairs/teams/vfs-10-10?

**Review by JoshingYou1 — APPROVED** [review]

File: app/models/health_care_application.rb [hunk 2]
```diff
@@ -262,4 +266,11 @@ def log_submission_failure
   def send_failure_mail
     HCASubmissionFailureMailer.build(parsed_form['email'], google_analytics_client_id).deliver_now
   end
+
+  # rubocop:disable Lint/UselessAssignment
+  def override_submission_params
+    override_parser = HCA::OverridesParser.new(parsed_form)
+    parsed_form = override_parser.override
```
Comment (lihanli) [inline app/models/health_care_application.rb:273]: `parsed_form` is a method, you can't assign it a new value

File: app/models/health_care_application.rb [hunk 3]
```diff
@@ -30,6 +30,10 @@ class HealthCareApplication < ApplicationRecord
     hca.saved_change_to_attribute?(:state) && hca.failed?
   end)
 
+  before_save do
```
Comment (lihanli) [inline app/models/health_care_application.rb:33]: the model isn't saved when the form is submitted synchronously

**Comment by lihanli** [comment]
you should add a test that tests whether or not the form is actually transformed during both sync and async submission, that test will tell you if your code is working

File: app/services/hca/overrides_parser.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    def override
+      override_address_states
+
+      form
+    end
+
+    def override_address_states
+      override_individual_address('veteranAddress')
+      override_individual_address('spouseAddress')
+    end
```
Comment (lihanli) [inline app/services/hca/overrides_parser.rb:57]: `veteranHomeAddress` should be overridden too

**Review by JoshingYou1 — APPROVED** [review]

**Review by rmtolmach — APPROVED** [review]

**Review by lihanli — APPROVED** [review]

**Pull request merged**